### PR TITLE
Refactor write api authorisation (+minor)

### PR DESCRIPTION
**Pull request by GRyall**
~depends on #180~
* Slight refactor of write API to reduce repeated code in relation to
authorisation
* improve error messages and response codes
* Fix minor error in text of error message in write API

### CODE REVIEW

**Comment by gregcorbett** [comment]
#180 has now been merged. @GRyall can this be rebase'd and the conflict resolved?

Thanks
Greg

**Comment by GRyall** [comment]
> 
> 
> #180 has now been merged. @GRyall can this be rebase'd and the conflict resolved?
> 
> Thanks
> Greg

@gregcorbett done

File: htdocs/PI/write/PIWriteRequest.php [hunk 1]
```diff
@@ -569,6 +569,17 @@ private function checkUserAuthenticated () {
       }
     }
 
+    private function checkAuthorisation (Site $siteService, \Site $site, $identifier, $indentifierType) {
+      try {
+        $siteService->checkAuthroisedAPIIDentifier($site, $identifier, $indentifierType);
```
Comment (jrha) [inline htdocs/PI/write/PIWriteRequest.php]: I've opened an issue around the naming of this function, but you didn't touch it here, so no need to fix it now.

**Comment by jrha** [comment]
I finished refactoring the entire code

File: htdocs/PI/write/PIWriteRequest.php [hunk 2]
```diff
@@ -569,6 +569,17 @@ private function checkUserAuthenticated () {
       }
     }
 
+    private function checkAuthorisation (Site $siteService, \Site $site, $identifier, $indentifierType) {
```
Comment (gregcorbett) [inline htdocs/PI/write/PIWriteRequest.php]: Whhy does the second `Site` have a `\`, but the first doesn't?

**Comment by gregcorbett** [comment]
@GRyall, other than my one comment I am happy with this PR.

File: htdocs/PI/write/PIWriteRequest.php (hunk 2 above)
Comment (GRyall) [reply htdocs/PI/write/PIWriteRequest.php]: https://www.php.net/manual/en/language.namespaces.php Sorry

Comment (gregcorbett) [reply htdocs/PI/write/PIWriteRequest.php]: at the hackathon: 
> George: "it's to do with namespaces, the first Site object is actually a SiteService and we are in the right namespace for that, the second Site is an actual Site object, and we aren't in the  namespace for that..."
> Greg: *wails in pain*

**Pull request merged**